promoter/promoter.go

```diff
@@ -164,3 +182,71 @@ func (p *Promoter) staticAddrDiff(ctx context.Context) (toAdd []WatchedAddress,
 	}
 	return
 }
+
+// threadedCreditTransactions continuously polls the db for uncreditted txns and
+// notifies the credit promoter about them.
+func (p *Promoter) threadedCreditTransactions() {
+	// TODO: Fetch all txns with credited == false.
+
+	// TODO: Credit the promoter. NOTE: This needs to be idempotent so we
+	// include a txn ID in the request. Requesting a credit with the same
+	// txn ID should still only credit it once.
+
+	// TODO: Update the credited txns in the database as credited = true.
+}
+
+// threadedPollTransactions continuously polls skyd for transactions related to
+// watched addresses and writes them to the DB.
+func (p *Promoter) threadedPollTransactions() {
+	t := time.NewTicker(txnPolInterval)
+	defer t.Stop()
+	for {
+		select {
+		case <-p.staticBGCtx.Done():
+			return
+		case <-t.C:
+		}
+		p.staticLogger.WithTime(time.Now()).Info("Starting to poll transactions from skyd")
+
+		// Get used addresses.
+		c, err := p.staticColWatchedAddresses().Find(p.staticBGCtx, bson.M{
+			"$or": bson.A{
+				bson.M{"user_id": bson.M{"$exists": true}},
+				bson.M{"user_id": bson.M{"$ne": ""}},
+			},
+		})
+		if err != nil {
+			p.staticLogger.WithError(err).Error("Failed to fetch used addresses")
+			continue
+		}
+
+		// For each one get the related txns from skyd and save them to
+		// the db.
+		var nAddresssInserted, nTxnsInserted int
+		for c.Next(p.staticBGCtx) {
+			// Get address.
+			var wa WatchedAddress
```

Comment:
We could. It's not too bad because it's stack allocated but definitely possible.
I would only optimise like that if a profile clearly shows this is an issue though because it's just a bit safer to make a new one and pass that as a reference than reusing one var.